newdle/client/src/auth.js

```diff
@@ -63,6 +69,25 @@ export function useAuthentication() {
     };
   }, [dispatch]);
 
+  useEffect(() => {
+    if (isUserLoggedIn) {
+      return;
+    }
+    const interval = window.setInterval(() => {
```

Comment:
Maybe we don't need to do this? Is it really a big deal if we leave the `windowId` in the state when the user closes the window? What's the worst thing that can happen? The user opening another window and the value being rewritten? Or maybe there's something I'm not seeing...

Reply:
`onunload` for windows is apparently not very reliable (at least based on google/SO results, didn't test it myself), but I need to know when the window closed (e.g. without logging in) so I can dispatch the action to hide the "logging in" dimmer/loader

Reply:
btw i just gave this a try, and it's as awful as people on SO said: `onunload` triggers once on load (for `about:blank` unloading), and even if i register onunload again while handling that event, it won't run when the window is actually closed :man_facepalming: